libnymea-core/debugserverhandler.cpp

```diff
@@ -24,15 +24,578 @@
+
+    m_pingReply->setPayload(processOutput);
+    m_pingReply->setHttpStatusCode(HttpReply::Ok);
+    m_pingReply->finished();
+    m_pingReply = nullptr;
+
+    m_pingProcess->deleteLater();
+    m_pingProcess = nullptr;
+}
+
+void DebugServerHandler::onDigProcessFinished(int exitCode, QProcess::ExitStatus exitStatus)
+{
+    qCDebug(dcDebugServer()) << "Dig process finished" << exitCode << exitStatus;
+    QByteArray processOutput = m_digProcess->readAll();
+    qCDebug(dcDebugServer()) << "Dig output:" << endl << qUtf8Printable(processOutput);
+
+    m_digReply->setPayload(processOutput);
+    m_digReply->setHttpStatusCode(HttpReply::Ok);
+    m_digReply->finished();
+    m_digReply = nullptr;
+
+    m_digProcess->deleteLater();
+    m_digProcess = nullptr;
+}
+
+void DebugServerHandler::onTracePathProcessFinished(int exitCode, QProcess::ExitStatus exitStatus)
+{
+    qCDebug(dcDebugServer()) << "Tracepath process finished" << exitCode << exitStatus;
+    QByteArray processOutput = m_tracePathProcess->readAll();
+    qCDebug(dcDebugServer()) << "Tracepath output:" << endl << qUtf8Printable(processOutput);
+
+    m_tracePathReply->setPayload(processOutput);
+    m_tracePathReply->setHttpStatusCode(HttpReply::Ok);
+    m_tracePathReply->finished();
+    m_tracePathReply = nullptr;
+
+    m_tracePathProcess->deleteLater();
+    m_tracePathProcess = nullptr;
+}
+
+void DebugServerHandler::onDebugReportGeneratorFinished(bool success)
 {
+    DebugReportGenerator *debugReportGenerator = static_cast<DebugReportGenerator *>(sender());
+    qCDebug(dcDebugServer()) << "Report generation finished" << (success ? "successfully" : "with error") << debugReportGenerator->reportFileName();
+    HttpReply *httpReply = m_runningReportGenerators.take(debugReportGenerator);
+
+    if (success) {
+        QVariantMap reportInformation;
+        reportInformation.insert("fileName", debugReportGenerator->reportFileName());
+        reportInformation.insert("fileSize", debugReportGenerator->reportFileData().size());
+        reportInformation.insert("md5sum", debugReportGenerator->md5Sum());
+        httpReply->setHttpStatusCode(HttpReply::Ok);
+        httpReply->setHeader(HttpReply::ContentTypeHeader, "application/json; charset=\"utf-8\";");
+        httpReply->setPayload(QJsonDocument::fromVariant(reportInformation).toJson(QJsonDocument::Indented));
+
+        m_finishedReportGenerators.insert(debugReportGenerator->reportFileName(), debugReportGenerator);
+    } else {
+        httpReply->setHttpStatusCode(HttpReply::InternalServerError);
+    }
+
+    httpReply->finished();
+}
 
+void DebugServerHandler::onDebugReportGeneratorTimeout()
+{
+    DebugReportGenerator *debugReportGenerator = static_cast<DebugReportGenerator *>(sender());
+    qCWarning(dcDebugServer()) << "Report generation timeouted. Cleaning up" << debugReportGenerator->reportFileName();
+    if (m_finishedReportGenerators.values().contains(debugReportGenerator)) {
```

Comment:
shouldn't this rather be m_runningReportGenerators instead? And also clean up some processes and whatnot?